iosHyperskillApp/iosHyperskillApp/Sources/Modules/StepQuizHints/Views/StepQuizHintCardView.swift

```diff
@@ -0,0 +1,134 @@
+import shared
+import SwiftUI
+
+extension StepQuizHintCardView {
+    struct Appearance {
+        let authorAvatarWidthHeight: CGFloat = 20
+        let reactionImageWidthHeight: CGFloat = 16
+    }
+}
+
+struct StepQuizHintCardView: View {
+    private(set) var appearance = Appearance()
+
+    let authorAvatarSource: String?
+
+    let authorName: String
+
+    let hintText: String
+
+    let hintHasReaction: Bool
+
+    let onHintReactionButtonTap: (ReactionType) -> Void
+
+    let onHintReportButtonTap: () -> Void
+
+    let onNextHintButtonTap: (() -> Void)?
+
+    @State private var isPresentingReportAlert = false
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: LayoutInsets.defaultInset) {
+            HStack {
+                LazyAvatarView(authorAvatarSource)
+                    .frame(widthHeight: appearance.authorAvatarWidthHeight)
+
+                Text(authorName)
+                    .font(.caption)
+                    .foregroundColor(Color(ColorPalette.black900Alpha38))
+
+                Spacer()
+
+                if !hintHasReaction {
+                    Button(Strings.StepQuiz.Hints.reportButton) {
+                        isPresentingReportAlert = true
+                    }
+                    .font(.subheadline)
+                    .foregroundColor(Color(ColorPalette.primaryAlpha60))
+                }
+            }
+
+            Text(hintText)
+                .font(.subheadline)
+                .foregroundColor(.primary)
+
+            if hintHasReaction {
+                if let onNextHintButtonTap = onNextHintButtonTap {
+                    StepQuizShowHintButton(text: Strings.StepQuiz.Hints.seeNextHint) {
+                        onNextHintButtonTap()
+                    }
+                } else {
+                    Text(Strings.StepQuiz.Hints.lastHint)
+                        .font(.caption)
+                        .foregroundColor(Color(ColorPalette.black900Alpha60))
```

Comment:
В дизайне используется цвет `colorOnSurface_0.6` и у нас есть `Color.secondaryText`.
Можно заменить на:
```suggestion
                        .foregroundColor(.secondaryText)
```

Reply:
И везде такой цвет можно заменить